### PR TITLE
Dev prav

### PR DESCRIPTION
**Pull request by pravrai**
Pushing my changes to master so we can build a master scene with our proof of concept implementations

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/FrozenMetalStudios/Soulless/pull/5" target="_blank">ReviewNinja

**Comment by DanApperloo** [comment]
God. The meta files really bloat the shit out of this review. Makes it hard to find your code changes

File: Project/Assets/Scripts/Player/PlayerHealth.cs [hunk 1]
```diff
@@ -48,6 +54,7 @@ public class PlayerHealth : MonoBehaviour {
 
     public void TakeDamage(int amount)
     {
+        print("player taking damage: " + amount);
```
Comment (DanApperloo) [inline Project/Assets/Scripts/Player/PlayerHealth.cs:57]: !fix Would be nice to see this using our logger :)

File: Project/Assets/Scripts/Player/PlayerAttack.cs [hunk 2]
```diff
@@ -69,30 +72,38 @@ public class PlayerAttack : MonoBehaviour {
...
-    IEnumerator CooldownHandler(Abilities ability)
+    //Coroutine used for ability cooldown
+    IEnumerator CooldownHandler(AbilityTest ability)
     {
         //print(ability.InputTag + " on " + ability.CoolDown+" second cooldown");
-        ability.isOffCooldown = false;
-        yield return new WaitForSeconds(ability.CoolDown);
-        ability.isOffCooldown = true;
+        ability.offCooldown = false;
+        yield return new WaitForSeconds(ability.cooldown);
```
Comment (DanApperloo) [inline Project/Assets/Scripts/Player/PlayerAttack.cs:105]: Can you double check for me this function can handle fraction of seconds? I'm looking at the documentation right now and it seems to, but just confirm for me :)

File: Project/Assets/Scripts/Player/PlayerAttack.cs [hunk 3]
```diff
@@ -1,66 +1,69 @@
 ﻿using UnityEngine;
 using System.Collections;
-using PlayerAbilities;
-
+using PlayerAbilityTest;
+using Assets.Scripts.Utility.Timers;
```
Comment (DanApperloo) [inline Project/Assets/Scripts/Player/PlayerAttack.cs:4]: A space beneath the using and comment would help visually

File: Project/Assets/Scripts/Player/PlayerAttack.cs [hunk 4]
```diff
@@ -69,30 +72,38 @@ public class PlayerAttack : MonoBehaviour {
...
-        if (ability.isOffCooldown)
+        //Check to see ability is off cooldown
+        if (ability.offCooldown && (player.playerHUD.energySlider.value - ability.energy) > 0)
         {
-            anim.Play(ability.InputTag, 0);
+            //play the correct animation
+            anim.Play(ability.animationTag, 0);
+            //set the correct trigger
             meleeAttackTrigger.enabled = true;
-            meleeAttackTrigger.SendMessage("updateDamage", ability.Damage);
```
Comment (DanApperloo) [inline Project/Assets/Scripts/Player/PlayerAttack.cs:79]: Might be nice to see this message as a STATIC type somewhere (would define in c, but not c#) however, I'm flexable on this issue.

Pros: Reusable. If name changes, don't have to look everywhere for it.
Cons: String is a little obscured. Little more overhead.

File: Project/Assets/Scripts/CombatManager.cs [hunk 5]
```diff
@@ -1,10 +1,14 @@
 ﻿using UnityEngine;
 using System.Collections;
 
+//Combat Manager
```
Comment (DanApperloo) [inline Project/Assets/Scripts/CombatManager.cs:4]: The usefulness of this class is:
Allows us to filter damage transactions at a global level
Allows us to log all damage transactions.

Right now this class is a little thin, however, If we design around its useful points, I think we can develop a system that provides greater runtime visibility and functionality

**Comment by DanApperloo** [comment]
Really nice Prav. Just a few small general comments. Some of the class bracketing is off. 
Some additional logging would be nice :)
Ping me for questions

File: Project/Assets/Scripts/CombatManager.cs (hunk 5 above)
Comment (DanApperloo) [inline Project/Assets/Scripts/CombatManager.cs:4]: !fixed

File: Project/Assets/Scripts/Player/PlayerAttack.cs (hunk 2 above)
Comment (pravrai) [inline Project/Assets/Scripts/Player/PlayerAttack.cs:105]: Yes fraction of seconds do work.

File: Project/Assets/Scripts/Player/PlayerAttack.cs (hunk 4 above)
Comment (pravrai) [inline Project/Assets/Scripts/Player/PlayerAttack.cs:79]: Would we want to centralize all the static types?,(have a single static class that contains these Static types?) or just have the associated static type be in the associated class (have a static type in MeleeAttackTrigger script with the string value) ?

**Pull request merged**